src/features/student/views/StudentToolsTracesView/components/StudentDetailedTraceAssociateModal/StudentDetailedTraceAssociateModal.test.ts

```diff
@@ -0,0 +1,209 @@
+      isLoading: { type: [Boolean, Object] as any, default: false },
+      onClick: { type: Function, default: undefined },
+    },
+    setup (props) {
+      const toBool = (v: any) =>
+        v && typeof v === 'object' && 'value' in v ? !!v.value : !!v
+      const disabledBool = computed(() => toBool(props.disabled))
+      const loadingBool = computed(() => toBool(props.isLoading))
+      const handleClick = () => {
+        if (!disabledBool.value && !loadingBool.value) {
+          props.onClick?.()
+        }
+      }
+      return { disabledBool, loadingBool, handleClick }
+    },
+    template: `
+      <button
+        class="av-button-stub"
+        :disabled="disabledBool || loadingBool"
+        @click="handleClick"
+      >
+        {{ label }}
+      </button>
+    `,
+  },
+  AvListItem: { name: 'AvListItem', template: '<div class="av-listitem-stub"><slot/></div>' },
+  AssociateSkillAutocompleteField: {
+    name: 'AssociateSkillAutocompleteField',
+    props: ['form', 'fieldName', 'associationType'],
+    template: '<div class="associate-skill-autocomplete-field-stub"></div>',
+  },
+}
+
+const TestHost = {
+  components: { StudentDetailedTraceAssociateModal },
+  props: {
+    onConfirmAssociateTrace: { type: Function, required: true },
+    onClose: { type: Function, required: true },
+  },
+  setup () {
+    const trace = { id: 't-001', title: 'Ma super trace' } as any
+    const show = true
+    return { trace, show }
+  },
+  template: `
+    <StudentDetailedTraceAssociateModal
+      :trace="trace"
+      :show="show"
+      :on-confirm-associate-trace="onConfirmAssociateTrace"
+      :on-close="onClose"
+    />
+  `,
+}
+
+BddTest().given('StudentDetailedTraceAssociateModal', () => {
+  let wrapper: VueWrapper
+  const onConfirmAssociateTrace = vi.fn()
+  const onClose = vi.fn()
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    isFormValidRef.value = true
+    isSubmittingRef.value = false
+    formStoreRef.value = {
+      isValid: true,
+      isValidating: false,
+      isDirty: false,
+      values: { selectedAssociation: null },
+    }
+
+    wrapper = mountComponent(TestHost, {
+      props: { onConfirmAssociateTrace, onClose },
+      global: { stubs },
+      useI18n: true,
+      useTanstack: true,
+      usePinia: true,
+    })
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  BddTest().when('the modal is opened', () => {
+    BddTest().then('it renders header and trace title', () => {
+      expect(wrapper.find('.av-modal-stub').exists()).toBe(true)
+      expect(wrapper.text()).toContain('Ma super trace')
+    })
+
+    BddTest().then('it renders the AssociateSkillAutocompleteField', () => {
+      const child = wrapper.findComponent({ name: 'AssociateSkillAutocompleteField' })
+      expect(child.exists()).toBe(true)
+    })
+  })
+
+  BddTest().when('footer primary button is displayed', () => {
+    BddTest().then('it is enabled when form is valid and not submitting', async () => {
```

Comment:
Rajouter des BddTest().and(...) pour chacun de tes if

Par exemple ici :

```suggestion
    BddTest().and('form is valid and not submitting', () => {
      BddTest().then('the submit button should be enabled', async () => {
```